DigitalLearningSolutions.Data/Services/UserService.cs

```diff
@@ -0,0 +1,70 @@
+﻿namespace DigitalLearningSolutions.Data.Services
+{
+    using System.Data;
+    using System.Linq;
+    using Dapper;
+    using DigitalLearningSolutions.Data.Models.User;
+
+    public interface IUserService
+    {
+        public User? GetUserByEmailAddress(string emailAddress);
+        public AdminUser? GetAdminUserByEmailAddress(string emailAddress);
+        public DelegateUser? GetDelegateUserByEmailAddress(string emailAddress);
+
+    }
+
+    public class UserService : IUserService
+    {
+        private readonly IDbConnection connection;
+
+        public UserService(IDbConnection connection)
+        {
+            this.connection = connection;
+        }
+
+        public User? GetUserByEmailAddress(string emailAddress)
+        {
+            // Assume user is a delegate (in Candidates Table)
+            User? user = GetDelegateUserByEmailAddress(emailAddress);
+
+            // If no Candidates entry found, look to AdminUsers
+            user ??= GetAdminUserByEmailAddress(emailAddress);
```

Comment:
I actually had a minor preference doing it the other way around (admin first, then delegate), because admin feels like the users "main" account, but tbh there will be more delegates so I actually think your approach is better! :)